provisioner/template.go

```diff
@@ -283,6 +284,10 @@ func list(args ...interface{}) []interface{} {
 	return args
 }
 
+func strAppend(list []string, item string) []string {
+	return append(list, item)
```

Comment:
Here we may want to clone the list first to not accidentally modify it.

Reply:
Doesn't `append` make a copy?

https://go.dev/play/p/_EM10XVQVn-

Reply:
Doesn't `append` make a copy?

https://go.dev/play/p/J8LQXgGgBxB

Reply:
Only when there is no capacity left

Reply:
OTOH it does exactly what append does so maybe it OK.